storage-api/src/java/org/apache/hadoop/hive/common/io/encoded/EncodedColumnBatch.java

```diff
@@ -78,13 +78,13 @@ public void setIndexBaseOffset(int indexBaseOffset) {
 
     @Override
     public String toString() {
-      String bufStr = "";
+      StringBuffer bufStr = new StringBuffer();
       if (cacheBuffers != null) {
         for (MemoryBuffer mb : cacheBuffers) {
-          bufStr += mb.getClass().getSimpleName() + " with " + mb.getByteBufferRaw().remaining() + " bytes, ";
+          bufStr.append(mb.getClass().getSimpleName() + " with " + mb.getByteBufferRaw().remaining() + " bytes, ");
```

Comment:
I think it's a little bit more than this.  There shouldn't be any "+" concatenation.  Should all be `.append()`

I already proposed to addressed this issue (HIVE-23540), but happy to defer to this PR
https://issues.apache.org/jira/browse/HIVE-23540

Reply:
Updated